src/unix/internal.h

```diff
@@ -495,4 +499,13 @@ int uv__get_constrained_cpu(uv__cpu_constraint* constraint);
 #endif
 #endif
 
+/* EVFILT_USER is available since OS X 10.6, DragonFlyBSD 4.0,
+ * FreeBSD 8.1, and NetBSD 10.0.
```

Comment:
Nit: libuv normally does not allow compile-time feature detection, as it tends to be unreliable for users. Most of those are EOL, except NetBSD 10.0 which is just released.

Could you change this to add runtime detection (of whether EVFILT_USER succeeded)?

Reply:
AFAIK, it seems to me that we've broken that rule many times, cuz you know, there are macros detecting the presence of system calls everywhere in `libuv`.

But your suggestion also sounds rational to me, I think it's doable. Are you leaning towards me utilizing the function like `check_c_source_runs` of `cmake`, or just writing some testing code inside the `uv__kqueue_init()` in `kqueue.c`?

Reply:
Is there? I thought it was usually detecting systems not headers, and then doing runtime detection of ENOSYS. Irregardless, it should not be a cmake check, that would be even worse than this current check